src/Utils.cc

```diff
@@ -252,4 +252,51 @@ void* fbgemmAlignedAlloc(
   return aligned_mem;
 }
 
+thread_type_t
+fbgemmGetThreadPartition(int m, int n, int thread_id, int num_threads) {
+  int m_num_threads;
+  assert(num_threads >= 1);
+  // Heuristic for determine the thread partitions for parallelizing across m or
+  // n dimensions.
+  // TODO: more smart ways for thread partitions considering the
+  // grain size (MR, NR) parameters
+  if (m < n && m / num_threads < 32 &&
```

Comment:
Why do we need this special case? Even when this if condition is met, we still want to parallelize across m when possible as long as m assigned for each thread is not much smaller than n assigned for each thread?

Reply:
This case is for purely parallelizing the n dimension, which is needed for the language models (where m << n, k).

Reply:
Update the heuristic algorithms for allocating the number of threads along the m and n dimensions, and removed this special case